python/tvm/script/parser.py

```diff
@@ -440,7 +443,13 @@ def check_decorator(decorators: List[ast.Expr]) -> bool:
         # add parameters of function
         for arg in node.params:
             arg_var = tvm.te.var(arg.name, self.parse_type(arg.ty, arg))
-            self.context.update_symbol(arg.name, arg_var, node)
+            # Note that this case is for T.match_buffer syntax sugar
+            if isinstance(arg.ty, (ast.TypeCall, ast.TypeApply)):
+                buf = self.transform(arg.ty)
```

Comment:
if this if statement only works for `T.match_buffer` it would be great to add some assertions (otherwise naming the return value `buf` here is confusing)

Reply:
fixed.